.github/workflows/codeql-build.yml

```diff
@@ -40,91 +48,116 @@ jobs:
+        working-directory: ../cFS
+
+      - name: Prep Build
+        run: ${{ inputs.make-prep }}
+        working-directory: ../cFS
+
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v1
         with:
           languages: c
           config-file: nasa/cFS/.github/codeql/codeql-security.yml@main
-          
-      - name: Copy sample_defs
-        run: ${{ inputs.setup }}
 
-      - name: Make prep
-        run: ${{ inputs.make-prep }}
-      
-      - name: Make Install
+      - name: Build
         run: ${{ inputs.make }}
+        working-directory: ../cFS 
 
-      - name: Run tests
-        run: ${{ inputs.tests }}
+      # - name: Run tests
+      #   run: ${{ inputs.tests }}
 
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@v1
-        
+        with:
+          add-snippets: true 
+
   CodeQL-Coding-Standard-Build:
     #Continue if check-for-duplicates found no duplicates. Always runs for pull-requests.
     needs: check-for-duplicates
     if: ${{ needs.check-for-duplicates.outputs.should_skip != 'true' }}
     runs-on: ubuntu-18.04
     timeout-minutes: 15
 
+    permissions:
+      security-events: write
     steps:
-      # Checks out a copy of your repository
-      - name: Checkout code
+      # Checks out a copy of calling repository
+      - name: Checkout Main Repo
         uses: actions/checkout@v2
-        with:
-          repository: nasa/cFS
-          submodules: true
 
-      - name: Check versions
+      - name: Clone cFS bundle
         run: |
-           git log -1 --pretty=oneline
-           git submodule
-      - name: Checkout codeql code      
-        uses: actions/checkout@v2
-        with:
-          repository: github/codeql 
-          submodules: true 
-          path: codeql
+          cd ..
+          git clone https://github.com/nasa/cFS.git --recurse-submodules
+          cd cFS
+          git log -1 --pretty=oneline
+          git submodule
+          rm -r .git
+
+      - name: Create symlink to current repo
+        run: |
+          cd ../cFS
+          rm -r ${{ inputs.component-path }} 
+          ln -s ${{github.workspace}} ${{ inputs.component-path }} 
+
+        # Setup the build system
+      - name: cFS Build Setup
+        run: ${{ inputs.setup }}
+        working-directory: ../cFS
+
+      - name: Prep Build
+        run: ${{ inputs.make-prep }}
+        working-directory: ../cFS
 
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v1
         with:
           languages: c
           config-file: nasa/cFS/.github/codeql/codeql-coding-standard.yml@main
 
-      - name: Copy sample_defs
-        run: ${{ inputs.setup }}
-
-      - name: Make prep
-        run: ${{ inputs.make-prep }}
-
-      - name: Make Install
+      - name: Build
         run: ${{ inputs.make }}
+        working-directory: ../cFS 
 
-      - name: Run tests
-        run: ${{ inputs.tests }}
+      # - name: Run tests
```

Comment:
@ArielSAdamsNASA, what is the use case for including tests in the CodeQL workflow?

Reply:
That can be removed

Reply:
If we don't run the unit tests here, where do they get run?

Reply:
Note I just went through a significant effort to fix a large number of uninitialized variables being used in app unit tests, something that I'd expect analysis tools to catch early and avoid.  I'm in favor of doing analysis on the unit tests, since the sort of warnings/errors that show up really should be addressed (earlier in the unit test development phase the better).

Reply:
I think as long as we build the tests the analysis tools should catch problems with them. My question was whether it makes sense to run them here or in a different workflow like https://github.com/nasa/cFS/blob/main/.github/workflows/build-cfs.yml

Reply:
Copy and concur!  As long as they still get analyzed, and the tests get run somewhere in CI I'm happy!